staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -381,6 +382,16 @@ func (t *tracker) add(gvr schema.GroupVersionResource, obj runtime.Object, ns st
 		}
 		if oldMeta.GetNamespace() == newMeta.GetNamespace() && oldMeta.GetName() == newMeta.GetName() {
 			if replaceExisting {
+				// If objects are equal, skip saving into store. This will not
+				// trigger Update event and return original data.
+				// For simplicity, we use reflect.DeepEqual to compare objects
+				// instead of comparing after encoding with runtime.Codec.
+				// Note that reflect.DeepEqual does not distinguish nil with
+				// empty value, do not mix them if you test not sending update
+				// event case.
```

Comment:
nit: how about the following, I think it's a little bit more clear.
```
To mimic the real apiserver's behavior, skip sending Modified event to fake watch channels if the object doesn't change. Note that unlike the byte comparison applied in the real apiserver, the tracker applies reflect.DeepEqual, which regards non-nil empty slices and maps as NOT equal to the nil ones.
```

Reply:
Thanks! Updated.